### PR TITLE
[SDK] Fix listing validation for marketplace v2 & v3

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts [hunk 1]
```diff
@@ -172,15 +172,33 @@ export class MarketplaceV3DirectListings<TContract extends DirectListingsLogic>
     let batches = await getAllInBatches(
       start,
       end,
-      this.contractWrapper.readContract.getAllValidListings,
+      this.contractWrapper.readContract.getAllListings, // TODO: revert to previous setup
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts:175]: don't think we should do this... would reduce performance quite a bit for everyone just to cover an edge case. Would rather fix it in the contract. Are we sure it actually throws for marketplace v3 too?

File: packages/sdk/test/evm/marketplace-v3.test.ts [hunk 2]
```diff
@@ -371,7 +371,8 @@ describe("Marketplace V3", async () => {
       );
     });
 
-    it("should return only active listings", async () => {
+    // TODO: fix timestamps for hardhat tests
+    it.skip("should return only active listings", async () => {
```
Comment (joaquim-verges) [inline packages/sdk/test/evm/marketplace-v3.test.ts:375]: i would leave marketplace v3 untouched for now. i think the fix is different there

File: packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts (hunk 1 above)
Comment (kumaryash90) [reply packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts:175]: Yes it's the same issue for v3 as well. Yeah, I agree, fixing it in the contract would be more efficient.

**Review by joaquim-verges — APPROVED** [review]
